ember_debug/vendor/startup-wrapper.js

```diff
@@ -75,15 +72,25 @@ var EMBER_VERSIONS_SUPPORTED = {{EMBER_VERSIONS_SUPPORTED}};
               return this._super.apply(this, arguments);
             }
           });
-          // Boot the inspector (or re-boot if already booted, for example in tests)
-          Ember.EmberInspectorDebugger.set('_application', app);
-          Ember.EmberInspectorDebugger.set('owner', instance);
-          Ember.EmberInspectorDebugger.start(true);
+
+          if (!Ember.EmberInspectorDebugger._application) {
+            bootEmberInspector(instance);
+          }
         }
       });
     }
   });
 
+  function bootEmberInspector(appInstance) {
```

Comment:
@alexhancock I am concerned moving this boot stuff may have broken some apps. I need to investigate further, but inspecting https://shipshape.io no longer works with inspector 3.5.0.

Reply:
Shoot! I'm pretty busy today but I could look tomorrow or definitely Monday.

Any more context on why you think specifically it might be the moving of the boot logic? If so, it would be helpful in debugging.

Reply:
@alexhancock I haven't had a chance to look yet, but it worked on https://shipshape.io before I merged the last couple PRs, so thought this was a logical first place to look, since we moved the boot logic around, maybe it's happening at the wrong time for newer Ember versions or something.

Reply:
I dug into this just now. The boot logic actually seems OK. It ends up having more to do with how I changed the `uniqueId` logic in `ember_debug/port.js`

https://github.com/emberjs/ember-inspector/blob/master/ember_debug/port.js#L17

```
this.get('namespace._application.name');
```

... is not set to anything in the case of https://shipshape.io. We could do something like...

```
  uniqueId: computed('namespace._application.name', function() {
    return this.get('namespace._application.name') || `app${RANDOM_NUMBER}`;
  }),
```

... but it feels like a better idea to think about what it really should be if `namespace._application.name` is only going to be set sometimes.

Any ideas?

Reply:
I'm okay with `app${RANDOM_NUMBER}` for the moment. We can work out something better later.

Reply:
https://github.com/emberjs/ember-inspector/pull/929 should fix...